modules/navier_stokes/src/physics/WCNSFVTurbulencePhysics.C

```diff
@@ -0,0 +1,328 @@
+    std::string kernel_name = prefix() + "ins_momentum_mixing_length_reynolds_stress_";
+    if (_porous_medium_treatment)
+      kernel_name = prefix() + "pins_momentum_mixing_length_reynolds_stress_";
+
+    params.set<UserObjectName>("rhie_chow_user_object") = _flow_equations_physics->rhieChowUOName();
+    for (const auto dim_i : make_range(dimension()))
+      params.set<MooseFunctorName>(u_names[dim_i]) = _velocity_names[dim_i];
+
+    for (const auto d : make_range(dimension()))
+    {
+      params.set<NonlinearVariableName>("variable") = _velocity_names[d];
+      params.set<MooseEnum>("momentum_component") = NS::directions[d];
+
+      getProblem().addFVKernel(kernel_type, kernel_name + NS::directions[d], params);
+    }
+  }
+}
+
+void
+WCNSFVTurbulencePhysics::addFluidEnergyTurbulenceKernels()
+{
+  if (_turbulence_model == "mixing-length")
+  {
+    const std::string u_names[3] = {"u", "v", "w"};
+    const std::string kernel_type = "WCNSFVMixingLengthEnergyDiffusion";
+    InputParameters params = getFactory().getValidParams(kernel_type);
+    assignBlocks(params, _blocks);
+    params.set<MooseFunctorName>(NS::density) = _density_name;
+    params.set<MooseFunctorName>(NS::cp) = _fluid_energy_physics->getSpecificHeatName();
+    params.set<MooseFunctorName>(NS::mixing_length) = _mixing_length_name;
+    params.set<Real>("schmidt_number") = getParam<Real>("turbulent_prandtl");
+    params.set<NonlinearVariableName>("variable") =
+        _flow_equations_physics->getFluidTemperatureName();
+
+    for (const auto dim_i : make_range(dimension()))
+      params.set<MooseFunctorName>(u_names[dim_i]) = _velocity_names[dim_i];
+
+    if (_porous_medium_treatment)
+      getProblem().addFVKernel(
+          kernel_type, prefix() + "pins_energy_mixing_length_diffusion", params);
+    else
+      getProblem().addFVKernel(
+          kernel_type, prefix() + "ins_energy_mixing_length_diffusion", params);
+  }
+}
+
+void
+WCNSFVTurbulencePhysics::addScalarAdvectionTurbulenceKernels()
+{
+  if (_turbulence_model == "mixing-length")
+  {
+    const std::string u_names[3] = {"u", "v", "w"};
+    const std::string kernel_type = "INSFVMixingLengthScalarDiffusion";
+    InputParameters params = getFactory().getValidParams(kernel_type);
+    assignBlocks(params, _blocks);
+    params.set<MooseFunctorName>(NS::mixing_length) = _mixing_length_name;
+
+    for (const auto dim_i : make_range(dimension()))
+      params.set<MooseFunctorName>(u_names[dim_i]) = _velocity_names[dim_i];
+
+    const auto & passive_scalar_names = _scalar_transport_physics->getAdvectedScalarNames();
+    const auto & passive_scalar_schmidt_number =
+        getParam<std::vector<Real>>("passive_scalar_schmidt_number");
+    if (passive_scalar_schmidt_number.size() != passive_scalar_names.size())
+      paramError("passive_scalar_schmidt_number",
+                 "The number of Schmidt numbers defined is not equal to the number of passive "
+                 "scalar fields!");
+
```

Comment:
User should be able to define one Sc number for all species, e.g., delay neutron precursor families are genenrally assigned the same Sc number

Reply:
adding this as a new feature